pennylane/ops/functions/equal.py

```diff
@@ -188,3 +232,12 @@ def _equal_shadow_measurements(
     k_match = op1.k == op2.k
 
     return return_types_match and wires_match and H_match and k_match
+
+
+def _obs_comparison_data(obs):
+    if isinstance(obs, Hamiltonian):
+        obs.simplify()
+        obs = obs._obs_data()  # pylint: disable=protected-access
+    else:
+        obs = {(1, frozenset(obs._obs_data()))}  # pylint: disable=protected-access
```

Comment:
Could we just directly use `op1.compare(op2)` instead?

Reply:
Good point :D

Reply:
I think its clearer if there's one function per type, so I've split it into an _equal function for if op1 is a Hamiltonian, and a separate one if its a Tensor. The Hamiltonian version uses op1.compare